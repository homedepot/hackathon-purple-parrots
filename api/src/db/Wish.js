const mongoose = require('mongoose')

const Schema = mongoose.Schema

const wishSchema = new Schema({
  child: {
    name: String,
    hometown: String,
    illness: String,
    age: Number,
  },
  type: { type: String, enum: ['go', 'meet', 'be', 'see'] },
  details: String,
  sponsor: {
    name: String,
    logo: String,
    links: [String]
  },
},
{
  timestamps: true
},
)

module.exports = mongoose.model('wish', wishSchema)
