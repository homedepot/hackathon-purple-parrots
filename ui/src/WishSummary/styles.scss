@import '../styles/colors.scss';

#WishSummary {
    padding: 1em;
    background-color: $lighter-gray;
    position: absolute;
    box-sizing: border-box;
    width: 100%;
    height: 100%;
}
