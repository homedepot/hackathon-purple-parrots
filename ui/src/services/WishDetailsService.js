import axios from 'axios'

const expressDomain = process.env.REACT_APP_expressDomain || 'http://localhost:3002'

const getWishDetails = async (id) => {
  let response = await axios.get(`${expressDomain}/wishes/${id}`);
  return response.data;
}

const makeAWish = async(wish) => {
  let response = await axios.post(`${expressDomain}/wishes`, wish)
  return response.data;
}

export const getWishes = async () => {
  // TODO query param options
  const { data } = await axios.get(`${expressDomain}/wishes`)
  return data
}

export default {
  getWishDetails,
  makeAWish
}
