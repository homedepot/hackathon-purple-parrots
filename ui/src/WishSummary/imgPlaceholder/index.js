import React, { Component } from 'react'
import './styles.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCamera } from '@fortawesome/free-solid-svg-icons'

export default class ImgPlaceholder extends Component {

    render() {
        return (
            <div className="icon-container">
                <FontAwesomeIcon className="camera-icon" icon={faCamera} />
                <a href="/review">Add Image</a>
            </div>
        )
    }
}
