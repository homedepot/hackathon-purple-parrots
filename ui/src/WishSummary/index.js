import React, { Component } from 'react';
import Wish from './wish';
import WishHeader from './wishHeader';
import WishFilter from './wishFilter';
import './styles.scss'

export default class WishSummary extends Component {

    render() {
        return (
            <div className="wishSummary">
                <WishHeader />
                <WishFilter />
                <Wish />
                <Wish />
                <Wish />
            </div>
        )
    }
}
