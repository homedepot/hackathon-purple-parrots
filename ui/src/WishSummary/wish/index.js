import React, { Component } from 'react'
import './styles.scss';
import chevronRight from '../../assets/Chevron_Forward.png';
import kidImg from '../../assets/icn_To_Be_Astronaut_130x130.png'
import sponsorImg from '../../assets/icn_To_Be_Astronaut_130x130.png'
import ImgPlaceholder from '../imgPlaceholder';

export default class Wish extends Component {
    render() {
        return (
            <div className="wish">
                <div className="date">
                </div>
                {false ? <img src={kidImg} alt="child" /> :<ImgPlaceholder text="Add Image" />}
                
                <div>
                    <p>
                        <strong>Maggie</strong> - Age 6 from Smyrna
                    </p>
                    <span className="summary">
                        Maggie wishes to go to Disney land
                    </span>
                </div>
                {false ? <img src={sponsorImg} alt="Sponsor" className="sponsor" /> : <a href="/review">Add Sponsor</a>}
                <img src={chevronRight} alt="click to view wish details" className="chevron"/>
            </div>
        )
    }
}
