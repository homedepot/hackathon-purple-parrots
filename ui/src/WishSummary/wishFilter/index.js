import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCalendarAlt } from '@fortawesome/free-solid-svg-icons'
import './styles.scss'

export default class wishFilter extends Component {
    render() {
        return (
            <div className="wish-filter">
                <div className="wish-search-container">
                    <input className="wish-search" placeholder="Search by Name, Sponsor, Location and/or Date" />
                    <FontAwesomeIcon className="calendar-icon" icon={faCalendarAlt} />
                </div>
                <div className="">
                    <p>Filter by Wish Type</p>
                    <input type="checkbox" id="toGo" name="toGo" />
                    <label for="toGo">To Go</label>
                    <input type="checkbox" id="toMeet" name="toMeet" />
                    <label for="toGo">To Meet</label>
                    <input type="checkbox" id="toBe" name="toBe" />
                    <label for="toGo">To Be</label>
                    <input type="checkbox" id="toSee" name="toSee" />
                    <label for="toGo">To See</label>
                </div>
            </div>
        )
    }
}
