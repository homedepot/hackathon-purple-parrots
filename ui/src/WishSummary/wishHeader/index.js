import React, { Component } from 'react'
import makeAWishLogo from '../../assets/Logo_MakeWish.png'
import galaxyLogo from '../../assets/Logo_Galaxy.png'
import './styles.scss'

export default class WishHeader extends Component {
    render() {
        return (
            <div id="wishHeader">
                <div className="logo-container">
                    <img src={makeAWishLogo} className="makeAWishLogo" alt="make a wish logo" />
                    <img src={galaxyLogo} className="galaxyLogo" alt="galaxy logo" />
                </div>
                <div className="user-details">
                    <span>Mary</span> | 
                    <a href="#">Sign Out</a>
                </div>
            </div>
        )
    }
}
