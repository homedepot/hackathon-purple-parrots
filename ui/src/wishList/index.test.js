import WishSummary from './index'
import React from 'react'
import { shallow } from 'enzyme'
import { getWishes } from '../services/WishDetailsService'
import Wish from './wish'

jest.mock('../services/WishDetailsService')

describe('WishSummary tests', () => {
  const mockWishList = [
    {
      'child': {
        'name': 'Jerel Weber',
        'age': 5,
        'hometown': 'North Robertside',
        'illness': 'SMTP'
      },
      'sponsor': {
        'links': []
      },
      '_id': '5d0528c11170183ea576e3da',
      'type': 'be',
      'details': 'overriding calculating Shirt',
      '__v': 0
    },
    {
      'child': {
        'name': 'Meta Brakus',
        'age': 16,
        'hometown': 'Rauburgh',
        'illness': 'Exclusive'
      },
      'sponsor': {
        'links': []
      },
      '_id': '5d0528c11170183ea576e3db',
      'type': 'meet',
      'details': 'Officer application Borders',
      '__v': 0
    }
  ]

  it('snapshot', () => {
    const wrapper = shallow(<WishSummary />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should display list of wishes', async () => {
    getWishes.mockImplementation(() => mockWishList)

    const wrapper = await shallow(<WishSummary />)
    const wishes = wrapper.find(Wish)
    expect(wishes.length).toEqual(2)
  })

})
