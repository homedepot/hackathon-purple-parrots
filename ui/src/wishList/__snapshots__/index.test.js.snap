// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishSummary tests snapshot 1`] = `ShallowWrapper {}`;
